crispy_forms/tests/test_form_helper.py

```diff
@@ -870,6 +873,16 @@ def test_label_class_and_field_class_bs4_offset_when_horizontal():
     assert '<div class="offset-sm-3 offset-md-4 col-sm-8 col-md-6">' in html
     assert html.count("col-sm-8") == 7
 
+    # Test col-X pattern
+
+    form.helper.label_class = "col-5 col-lg-4"
+    form.helper.field_class = "col-7 col-lg-8"
+    html = render_crispy_form(form)
+
+    assert '<div class="form-group row">' in html
+    assert '<div class="offset-5 offset-lg-4 col-7 col-lg-8">' in html
+    assert html.count("col-7") == 7
```

Comment:
I wonder if we could do with a SampleForm with just one input in it. Is it important, in this case, that we're testing that we get the same answer for many different inputs (7 in this case, or is it?!) ? 

I've been pondering for a while about how we write some of these tests. I wonder if we can generate small enough html we can 'just' dump the whole thing into the test. It's then clearer what we're actually testing for. At the moment every time I look at these, I jump into the debugger to evaluate html. 

But maybe I'm doing it all wrong 🤷‍♂ (I don't know what I don't know....)

Reply:
Dumping the whole HTML in the test would be very sensitive to even a new blank space inserted. But in the meantime we could load the html string as an html object, and then dump it. It would be more robust solution to prevent regression but it would also take time to re-write all the test :-/

Reply:
Thanks for the insight 👍